src/lib/shopify-graphql.js

```diff
@@ -0,0 +1,189 @@
+import { ApolloError } from 'apollo-server-express';
+import axios from 'axios';
+import { getContributorInfo } from './github';
+
+/*
+ * @todo: handle the case where a Shopify customer already exists and we try to create one (lookup and associate existing user with prisma account)
+ * @todo: use email address associated with user's github account (account for this in UI also)
+ * @todo: add tests
+ */
+
+const SHOPIFY_DISCOUNT_CODES = [
+  {
+    code: 'BUILDWITHGATSBY',
+    // rename this shizz
+    threshold: 1,
+    tag: 'contributor'
+  },
+  {
+    code: 'LEVEL2',
+    threshold: 5,
+    tag: 'level2'
+  }
+];
+
+export const createShopifyCustomer = async ({
+  githubUsername,
+  email,
+  firstName,
+  acceptsMarketing
+}) => {
+  const { totalContributions } = await getContributorInfo(githubUsername);
+  const tags = getEarnedCodes(totalContributions).map(({ tag }) => tag);
+
+  // doesnt work when a customer already exists
+  // need to get ID and add to prisma
+  const mutation = `
+    mutation {
+      customerCreate(input: {
+        acceptsMarketing: ${acceptsMarketing}
+        firstName: "${firstName}"
+        email: "${email}"
+        tags: ${JSON.stringify(tags)}
+      }) {
+        customer {
+          id
+        }
+        userErrors {
+          field
+          message
+        }
+      }
+    }
+  `;
+
+  // TODO do we get error messages back from Shopify?
+  // TODO how do we look up a customer by email?
+  try {
+    const {
+      data: {
+        data: { customerCreate: response }
+      }
+    } = await axios({
+      method: 'post',
+      url: `https://${process.env.SHOPIFY_URI}/admin/api/graphql.json`,
+      headers: {
+        'Content-Type': 'application/graphql',
+        'X-Shopify-Access-Token': process.env.SHOPIFY_GRAPHQL_TOKEN
+      },
+      data: mutation
+    });
+
+    if (response.userErrors.length > 0) {
+      throw new Error(response.userErrors[0].message);
```

Comment:
Might be worth checking for multiple errors; this may be overkill given how small this API is, but it would also be nice if even our error messages carried the Gatsby voice.